arangod/Aql/QueryRegistry.cpp

```diff
@@ -706,12 +778,21 @@ QueryRegistry::QueryInfo::QueryInfo(ErrorCode errorCode, double ttl)
       _isTombstone(true) {}
 
 QueryRegistry::QueryInfo::~QueryInfo() {
+  auto const startDestructionTime = std::chrono::steady_clock::now();
+  using namespace std::chrono_literals;
   if (!_promise.isFulfilled()) {
     // we just set a dummy value to avoid abandoning the promise, because this
     // makes it much more difficult to debug cases where we _must not_ abandon a
     // promise
     _promise.setValue(std::shared_ptr<ClusterQuery>{nullptr});
   }
+  auto const diff = std::chrono::duration_cast<std::chrono::seconds>(
```

Comment:
If we want this time stamp to be taken after `_query` is destroyed, we probably have to reset the shared_ptr manually here. Otherwise it will only be destroyed after the destructor is finished, right?

Reply:
Ah, you are right, I missed this

Reply:
Note that this measurement might, in the (rare?) cases where the promise hasn't been fulfilled, also include the time for callbacks attached to the future corresponding `_promise`, see RestAqlHandler. That's probably OK because this taking long would also have led to similar problems, but we should be aware when interpreting logs later @neunhoef.